src/stores/DialogStore/DialogStore.ts

```diff
@@ -1,16 +1,58 @@
 import {TabId} from "@blueprintjs/core";
-import {action, makeObservable, observable} from "mobx";
+import {action, autorun, makeObservable, observable} from "mobx";
 
 import {FileInfoType} from "components";
 import {WorkspaceDialogMode} from "components/Dialogs/WorkspaceDialog/WorkspaceDialogComponent";
 import {Snippet} from "models";
 import {AppStore, SnippetStore} from "stores";
 
+export enum DialogId {
+    About = "about-dialog",
+    CatalogQuery = "catalogQuery-dialog",
+    Snippet = "snippet-dialog",
+    ExistingSnippet = "existing-snippet-dialog",
+    NewSnippet = "new-snippet-dialog",
+    Contour = "contour-dialog",
+    DistanceMeasure = "distanceMeasure-dialog",
+    ExternalPage = "externalPage-dialog",
+    FileBrowser = "fileBrowser-dialog",
+    FileInfo = "fileInfo-dialog",
+    Fitting = "fitting-dialog",
+    Layout = "saveLayout-dialog",
+    Preference = "preference-dialog",
+    Region = "region-dialog",
+    Stokes = "stokes-dialog",
+    Vector = "vector-dialog",
+    Workspace = "workspace-dialog",
+    ShareWorkspace = "shareWork-dialog",
+    Hotkey = "hotkey-dialog"
+}
+
+interface showDialogProps {
+    oldLayoutName?: string;
+    mode?: WorkspaceDialogMode;
+    url?: string;
+    title?: string;
+    snippet?: Snippet;
+    name?: string;
+}
+
 export class DialogStore {
     private static staticInstance: DialogStore;
 
+    @observable workspaceDialogMode = WorkspaceDialogMode.Hidden;
+    @observable selectedFileInfoDialogTab: TabId = FileInfoType.IMAGE_HEADER;
+    @observable externalPageDialogUrl: string;
+    @observable externalPageDialogTitle: string;
+    @observable dialogVisible = new Map<string, boolean>();
+
     constructor() {
         makeObservable(this);
+        autorun(() => {
```

Comment:
I think we can initialize it without `autorun`.

Reply:
okay.